influxql/functions.gen.go

```diff
@@ -377,6 +380,50 @@ func (r *FloatElapsedReducer) Emit() []IntegerPoint {
 	return nil
 }
 
+// FloatSampleReduces implements a reservoir sampling to calculate a random subset of points
+type FloatSampleReducer struct {
+	count int // how many points we've iterated over
+
+	points floatPoints // the reservoir
+}
+
+// NewFloatSampleReducer creates a new FloatSampleReducer
+func NewFloatSampleReducer(size int) *FloatSampleReducer {
+	return &FloatSampleReducer{
+		points: make(floatPoints, size),
+	}
+}
+
+// AggregateFloat aggregates a point into the reducer.
+func (r *FloatSampleReducer) AggregateFloat(p *FloatPoint) {
+	// Fill the reservoir with the first n points
+	if r.count < len(r.points) {
+		r.points[r.count] = *p
+		r.count++
+		return
+	}
+
+	// Generate a random integer between 1 and the count and
+	// if that number is less than the length of the slice
+	// replace the point at that index rnd with p.
+	rnd := rand.Intn(r.count + 1)
```

Comment:
Do you have an article or website that can be referenced to show that this produces two random points with only minimal bias from the pseudorandom generator? I just want to make sure this method doesn't have an accidental introduction of unnecessary bias.